vault/resource_consul_secret_backend_role.go

```diff
@@ -158,17 +170,38 @@ func consulSecretBackendRoleRead(d *schema.ResourceData, meta interface{}) error
 	}
 
 	data := secret.Data
-	d.Set("name", name)
+	if err := d.Set("name", name); err != nil {
+		return err
+	}
+	var pathKey string
 	if _, ok := d.GetOk("path"); ok {
-		d.Set("path", backend)
+		pathKey = "path"
 	} else {
-		d.Set("backend", backend)
+		pathKey = "backend"
+	}
+	if err := d.Set(pathKey, backend); err != nil {
+		return err
+	}
+
+	// map request params to schema fields
+	params := map[string]string{
+		"policies":     "policies",
+		"max_ttl":      "max_ttl",
+		"ttl":          "ttl",
+		"token_type":   "token_type",
+		"local":        "local",
+		"consul_roles": "roles",
+	}
+
+	for k, v := range params {
+		val, ok := data[k]
+		if k == "consul_roles" && !ok {
```

Comment:
Wondering if we should have a similar case for `policies`? Since according to the Vault docs `Either "policies"
or "consul_roles" are required for Consul 1.5 and above`. Are we only targeting Consul 1.4 for now (in which case this would make sense since only `policies` is required)

Reply:
I don't  believe so. This case is only for vault 1.9 and below, and is not tied a specific version of Consul. I suppose Vault will figure that out for us. Unless I am misunderstanding your question?

Reply:
What if in the future a user would like to only set `roles` and not `policies`? This is technically possible in Vault, but the user wouldn't be able to do that with the current implementation of the provider if I understand it correctly

Reply:
Actually never mind, I think in that case the Read function should just set the `policies` field to be empty?